### ISSUE TITLE
Kernel crash in 4 lines of simple c++17 code.

### ISSUE THREAD

**Issue by codecnotsupported** [QuantStack/xeus-cling#160]
```
#include <any>
#include <vector>
std::vector<int> a;
std::any_cast<std::vector<char>>(a)

```
Yes, this code does not make sense or is suppose to work, but it crashes the cling c++17 kernel. (somehow)

**Comment by SylvainCorlay** [comment]
Indeed, this code raises an standard exception (bad any cast), which we are catching properly, but the xeus kernel is not recovering correctly. The classic interpreter is not showing the type of exception but is recovering properly.

I need to investigate.
